File: internal/pkg/dto/response/ai_employee_preview.go
package response

type PreviewMountedSkill struct {
	ID   int64  `json:"id"`
	Name string `json:"name"`
}

type AIEmployeePreviewResponse struct {
	Content         string                   `json:"content"`
	ModelName       string                   `json:"modelName"`
	KnowledgeStatus string                   `json:"knowledgeStatus"`
	Sources         []CopilotKnowledgeSource `json:"sources"`
	MountedSkills   []PreviewMountedSkill    `json:"mountedSkills"`
	DurationMs      int64                    `json:"durationMs"`
}

File: web/lib/api/ai-employee-preview.ts
import { request } from "@/lib/api/client"
import type { CreateAIAgentPayload } from "@/lib/api/admin"

export type PreviewTurn = { role: "user" | "assistant"; content: string }
export type EmployeePreviewResult = {
  content: string
  modelName: string
  knowledgeStatus: "not_configured" | "empty" | "matched"
  sources: { knowledgeBaseId: number; documentId: number; chunkId: number; title: string; content: string }[]
  mountedSkills: { id: number; name: string }[]
  durationMs: number
}
export function previewAIEmployee(draft: CreateAIAgentPayload, messages: PreviewTurn[], signal?: AbortSignal) {
  return request<EmployeePreviewResult>("/api/dashboard/ai-agent/preview", {
    method: "POST", body: JSON.stringify({ draft, messages }), signal,
  })
}
